async_processor/kafka_pub_sub.py

```diff
@@ -0,0 +1,98 @@
+from contextlib import asynccontextmanager
+from typing import AsyncIterator, Optional
+
+from kafka import KafkaConsumer, KafkaProducer
+from starlette.concurrency import run_in_threadpool
+
+from async_processor.logger import logger
+from async_processor.types import (
+    Input,
+    InputFetchAckFailure,
+    InputMessageFetchFailure,
+    KafkaInputConfig,
+    KafkaOutputConfig,
+    Output,
+)
+
+
+class KafkaInput(Input):
+    def __init__(self, config: KafkaInputConfig):
+        self._bootstrap_servers = config.bootstrap_servers
+        self._topic_name = config.topic_name
+        self._auth = config.auth
+        self._consumer_group_id = config.consumer_group_id
+        self._consumer = KafkaConsumer(
+            self._topic_name,
+            bootstrap_servers=self._bootstrap_servers,
+            group_id=config.consumer_group_id,
+            sasl_plain_username=config.auth.username,
+            sasl_plain_password=config.auth.password,
+            security_protocol="SASL_SSL",
+            sasl_mechanism="PLAIN",
+            enable_auto_commit=False,
+        )
+        self._producer = KafkaProducer(
+            bootstrap_servers=self._bootstrap_servers,
+            sasl_plain_username=config.auth.username,
+            sasl_plain_password=config.auth.password,
+            security_protocol="SASL_SSL",
+            sasl_mechanism="PLAIN",
+        )
+
+    @asynccontextmanager
+    async def get_input_message(
+        self,
+    ) -> AsyncIterator[Optional[str]]:
+        while True:
```

Comment:
Please take a look at https://github.com/truefoundry/async_service/pull/11/files#diff-ed303578d39400a4b962213796a032dfa943554525a6e15cddb4dd1e6c8a766d to see the changes we have made in the structure of the `get_input_message` method implementation.